src/index.ts

```diff
@@ -76,6 +76,13 @@ function initConfig(projectConfig: Forceable<Config>):
       Constants.TRACE_SERVICE_LABEL_VALUE_LIMIT) {
     config.maximumLabelValueSize = Constants.TRACE_SERVICE_LABEL_VALUE_LIMIT;
   }
+  // Make rootSpanNameOverride a function if not already.
+  if (typeof config.rootSpanNameOverride === 'string') {
+    const spanName = config.rootSpanNameOverride;
+    config.rootSpanNameOverride = () => spanName;
+  } else if (typeof config.rootSpanNameOverride !== 'function') {
+    config.rootSpanNameOverride = (name: string) => name;
```

Comment:
AFAICT, there are no tests paths covering the `string` case.

Reply:
Thanks for pointing that out. I added a test to check that a string config options gets turned into a function when the configuration is getting normalized.

Reply:
TSC doesn't allow the field `rootSpanNameOverride` to be a string when accessed from a `Tracer` instance, so I am not covering that in this file.